engines/bastion_katello/app/assets/javascripts/bastion_katello/content-hosts/details/views/content-host-info.html

```diff
@@ -138,7 +138,7 @@ <h4 translate>System Purpose</h4>
 
         <dt translate>Add ons</dt>
         <dd bst-edit-multiselect="host.subscription_facet_attributes.purpose_addons"
-            readonly="denied('edit_hosts', host)"
+            readonly="denied('edit_hosts', host) || (organization.system_purposes.addons.length + host.subscription_facet_attributes.purpose_addons.length === 0)"
```

Comment:
the logic is bit complicated, any way to move it to the controller?

Reply:
I wanted to simply check `purposeAddons().length === 0` but since that controller method returns a promise it just doesn't work here. At least that's what I took away from it, so I consider what is here a fair tradeoff but not my ideal approach. Are you aware of a way that it might work? Definitely open to changing it

Reply:
you could try something like this:

In view:
```html
readonly="denied('edit_hosts', host) || noPurposeAddons(organization, host)"
```
In controller
```javascript
$scope.noPurposeAddons = function(organization, host) {
  return organization.system_purposes.addons.length + host.subscription_facet_attributes.purpose_addons.length === 0
}
```

Reply:
I tried something similar to no avail before I first pushed this up, and I tried this exactly just now because I think it's a good idea - but no joy:

```
Cannot read property 'system_purposes' of undefined
```

Each time this is called the organization hasn't been loaded yet. It made me wonder about how the current code on this PR is working. Either it is very lucky timing-wise (I haven't seen any errors whatsoever) or there's an angular nuance that is making it work.

Reply:
you may have to do:
```javascript
 return organization && host && organization.system_purposes.addons.length + host.subscription_facet_attributes.purpose_addons.length === 0
```

I've seen something similar, I'm not sure about the nuances of angular either, I think it's something to do with the way angular binds the variable to the $scope and evaluates it in the view :man_shrugging: 

Let me know if you have any luck with the above, if not, we can keep what you have!

Reply:
The similar approach I tried was this:

```
$scope.noPurposeAddons = function() {
  $scope.organization.$promise.then(function(org) {
    return org.system_purposes.addons.length + $scope.host.subscription_facet_attributes.purpose_addons.length === 0;
    });
  };
```

Reply:
> you may have to do:
> 
> ```js
>  return organization && host && organization.system_purposes.addons.length + host.subscription_facet_attributes.purpose_addons.length === 0
> ```
> 
> I've seen something similar, I'm not sure about the nuances of angular either, I think it's something to do with the way angular binds the variable to the $scope and evaluates it in the view 🤷‍♂️
> 
> Let me know if you have any luck with the above, if not, we can keep what you have!

Whoah, it might be working, gonna refresh a few dozen times ...